### ISSUE TITLE
Adding a non-associative operator for precedence climbing?

### ISSUE THREAD

**Issue by jason5122** [matthieucham/calculator#1]
Hi, thank you for implementing this calculator. I wanted to make my own version for first-order logic and I am wondering how I could add a non-associative operator. For example, `True → True → False` would not be a well-formed sentence in the grammar if `→` was non-associative. I've read some of http://www.engr.mun.ca/~theo/Misc/exp_parsing.htm but I'm still not clear on how to do this.

**Comment by matthieucham** [comment]
Hello, could you please provide some more example of what you're trying to achieve with this arrow operator ?
The associativity of each operator is managed already: it's left by default, but you can specify right like in the Pow operator

**Comment by jason5122** [comment]
Sure. According to [Wikipedia](https://en.wikipedia.org/wiki/Operator_associativity#Non-associative_operators), "non-associative operators are operators that have no defined behavior when used in sequence in an expression." I'm hoping to throw an error for statements such as `P → Q → R` or `P ∨ Q ∨ R`, saying these are ambiguous.

Java's operator precedence would be a pretty good example: http://web.deu.edu.tr/doc/oreily/java/langref/ch04_14.htm. An operator like `<` is non-associative since you cannot have multiple chained together, like `1 < 2 < 3`. However, `(1 < 2) < 3` and `1 < (2 < 3)` are well-formed.

Please let me know if you need any more clarification!

**Comment by matthieucham** [comment]
Ok I see, it's not that easy indeed. Sorry I don't have the time to help you atm, but I'm definitely interested in seeing your solution when you'll get one :)

**Comment by jason5122** [comment]
I found a solution! Prof. Norvell has another published [article on Pratt Parsing](https://www.engr.mun.ca/~theo/Misc/pratt_parsing.htm). He has some pseudocode for precedence climbing involving non-associative operators just before the Pratt Parsing section, which I implemented.

Here's something I quickly threw together, excuse the poor code quality. It's not meant to be used as a calculator unfortunately; it outputs a tree structure instead.

[calc_test.py.zip](https://github.com/matthieucham/calculator/files/6176977/calc_test.py.zip)
